functions/go/list-setters/listsetters/walk.go

```diff
@@ -0,0 +1,69 @@
+package listsetters
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// visitor is implemented by structs which need to walk the configuration.
+// visitor is provided to accept to walk configuration
+type visitor interface {
```

Comment:
this is the same as [create-setters](https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/create-setters/createsetters/walk.go) and [apply-setters](https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/apply-setters/applysetters/walk.go). Happy to refactor into a common pkg if that makes sense.

Reply:
Will do in a follow up PR as discussed offline.